### PR TITLE
Implement sparse pow.

### PR DESCRIPTION
**Pull request by ezyang**
Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: torch/lib/THCS/generic/THCSTensorMath.cu [hunk 1]
```diff
@@ -463,6 +463,38 @@ void THCSTensor_(cmul)(THCState *state, THCSTensor *r_, THCSTensor *t, THCSTenso
   THCTensor_(free)(state, s_values_);
 }
 
+#if defined(THCS_REAL_IS_FLOAT) || defined(THCS_REAL_IS_DOUBLE)
+void THCSTensor_(pow)(THCState *state, THCSTensor *r_, THCSTensor *t, real value) {
```
Comment (apaszke) [inline torch/lib/THCS/generic/THCSTensorMath.cu]: It seems that all pointwise functions will have nearly the same code. Maybe it's worth extracting to some helpers or macros?

Comment (ezyang) [inline torch/lib/THCS/generic/THCSTensorMath.cu]: Yes, that's the plan!

File: torch/sparse/__init__.py [hunk 2]
```diff
@@ -78,12 +78,6 @@ def __rsub__(self, other):
     def __matmul__(self, other):
         raise NotImplementedError
 
-    def __pow__(self, other):
```
Comment (apaszke) [inline torch/sparse/__init__.py]: Why not implement them instead of removing the methods?

Comment (ezyang) [inline torch/sparse/__init__.py]: I think I misunderstood how our codegen works. When `__pow__` was defined as NotImplemented, my tests failed; when I removed it started passing, so I assumed the parent class was magically defining __pow__. I guess I should look at this more.

Comment (apaszke) [inline torch/sparse/__init__.py]: Ah I didn't see that it was shadowing the method from `_TensorBase`, my bad.

**Review by apaszke — APPROVED** [review]

**Pull request merged**